crates/rpc-types/src/genesis.rs

```diff
@@ -27,3 +29,22 @@ pub struct OptimismBaseFeeInfo {
     /// EIP-1559 denominator after canyon
     pub eip1559_denominator_canyon: Option<u64>,
 }
+
+impl OptimismGenesisInfo {
+    /// Extract the Optimism genesis info from the genesis.
+    pub fn extract_from(genesis: &Genesis) -> Option<Self> {
```

Comment:
could we add another helper type that has both genesisInfo and feeinfo as fields and then impl fn extract_from(others: &Otherfields)?

and also `Tryfrom<Otherfields>` that simply calls extract?